### ISSUE TITLE
Hard Coded Activity-ID for bounces

### ISSUE THREAD

**Issue by pbatroff** [WeMoveEU/uk.co.compucorp.civicrm.mailjet#16]
In [Endpoint.php](https://github.com/WeMoveEU/uk.co.compucorp.civicrm.mailjet/blob/master/CRM/Mailjet/Page/EndPoint.php#L248) the Activity Id is hard coded to 58 at the moment. 

I assumed it is a default CiviCRM Activity, but in a fresh install it is not available. Is this a mistake on my end, or did it change recently and got removed? (Testing on 5.19.4).

Otherwise I would propose to either configure an activity-id, or let the Extension create on on installation and use that instead. (saving the ID to an extension setting).

**Comment by tttp** [comment]
Indeed, 58 is the hardcoded bounce activity (it always puzzled me that civimail by default doesn't keep a clear and visible history or why an email was put on hold)

Indeed, that would be a great improvement to create the activity type if not existing.
My only concern is that it's an hot path (lots of requests), so fetching the id either by the setting or by the name "Bounce" might slow down (it's already slow)

    {
            "id": "8191",
            "option_group_id": "2",
            "label": "Bounce",
            "value": "58",
            "name": "Bounce",
            "filter": "0",
            "is_default": "0",
            "weight": "58",
            "description": "Email was bounced",
            "is_optgroup": "0",
            "is_reserved": "0",
            "is_active": "1"
        }

**Comment by pbatroff** [comment]
Ok, I understand. How about a configuration file, where the ID is stored and can be read from fast? In that case, patching the source file is not needed.

**Comment by tttp** [comment]
like adding a constant BOUNCE_TYPE_ACTIVITY in the settings.civicrm.php and do BOUNCE_TYPE_ACTIVITY || 58 ?

**Comment by pbatroff** [comment]
Yeah, that sounds reasonable for me! If added to documentation then it makes a lot of sense!
